ethers-contract/ethers-contract-abigen/src/lib.rs

```diff
@@ -368,6 +396,11 @@ impl ContractBindings {
     }
 }
 
+/// Returns whether the current executable is a cargo build script.
+fn in_build_script() -> bool {
+    std::env::var("CARGO_MAKEFLAGS").is_ok()
```

Comment:
TIL

Reply:
honestly i just went over all of the env vars, and this one was the only one that is set only on build scripts, per docs. Not 100% sure

Reply:
`TARGET` should be fine as well.